File: BACKEND/src/main/java/com/model/NurseOff.java
package com.model;

import javax.persistence.*;

@Entity
public class NurseOff {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private OffType type;

    @Enumerated(EnumType.STRING)
    private OffStatus status;

    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Nurse nurse;

    public NurseOff()
    {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public OffType getType() {
        return type;
    }

    public void setType(OffType type) {
        this.type = type;
    }

    public OffStatus getStatus() {
        return status;
    }

    public void setStatus(OffStatus status) {
        this.status = status;
    }

    public Nurse getNurse() {
        return nurse;
    }

    public void setNurse(Nurse nurse) {
        this.nurse = nurse;
    }
}

File: FRONTEND/src/app/models/nurseOff.ts
import {Nurse} from './nurse';
import {OffType} from './offType';
import {OffStatus} from './offStatus';

export class NurseOff {
  id: number;
  type: OffType;
  status: OffStatus;
  nurse: Nurse;

  constructor(type: OffType, status: OffStatus, nurse: Nurse, id?: number) {
    this.type = type;
    this.status = status;
    this.nurse = nurse;
    this.id = id;

  }

}
